test/e2e/medusa_test.go

```diff
@@ -198,6 +198,40 @@ func checkPurgeCronJobExists(t *testing.T, ctx context.Context, namespace string
 	}, polling.medusaBackupDone.timeout, polling.medusaBackupDone.interval, "Medusa purge Job didn't finish within timeout")
 }
 
+func checkNoPurgeCronJob(t *testing.T, ctx context.Context, namespace string, dcKey framework.ClusterKey, f *framework.E2eFramework, kc *api.K8ssandraCluster) {
+	require := require.New(t)
+	// Get the Cassandra pod
+	dc1 := &cassdcapi.CassandraDatacenter{}
+	err := f.Get(ctx, dcKey, dc1)
+	// check medusa containers exist
+	require.NoError(err, "Error getting the CassandraDatacenter")
+	t.Log("Checking that all the Medusa related objects have been created and are in the expected state")
+	// check that the cronjob exists
+	cronJob := &batchv1.CronJob{}
+	err = f.Get(ctx, framework.NewClusterKey(dcKey.K8sContext, namespace, medusapkg.MedusaPurgeCronJobName(kc.SanitizedName(), dc1.SanitizedName())), cronJob)
+	require.NoErrorf(err, "Error getting the Medusa purge CronJob. ClusterName: %s, DatacenterName: %s", kc.SanitizedName(), dc1.SanitizedName())
+	require.Equal("k8ssandra-operator", cronJob.Spec.JobTemplate.Spec.Template.Spec.ServiceAccountName, "Service account name is not correct")
+	// create a Job from the cronjob spec
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      "test-purge-job",
+		},
+		Spec: cronJob.Spec.JobTemplate.Spec,
+	}
+	err = f.Create(ctx, dcKey, job)
+	require.NoErrorf(err, "A Medusa purge Job was created when it should not have been. ClusterName: %s, DataceneterName: %s, Namespace: %s, JobName: test-no-purge-job", kc.SanitizedName(), dc1.SanitizedName(), namespace)
+	// ensure the job was not created
+	require.Never(func() bool {
+		updated := &batchv1.Job{}
+		err := f.Get(ctx, framework.NewClusterKey(dcKey.K8sContext, namespace, "test-purge-job"), updated)
+		if err != nil {
+			return false
+		}
+		return updated.Status.Succeeded == 1
+	}, polling.medusaBackupDone.timeout, polling.medusaBackupDone.interval, "Timeout checking for Medusa purge Job")
```

Comment:
@adejanovski can you help me out with what timeout / interval to use here.  This is waiting an hour and failing.

Reply:
Just pushed a commit to rework this, let's see how CI goes.

Reply:
Looks like it's still timing out

Reply:
scratch that - the actual test is failing - err is nil 
https://github.com/k8ssandra/k8ssandra-operator/actions/runs/9194241488/job/25287268234?pr=1292#step:14:154

Reply:
actually - it's fine, the change I made to make the bool a pointer seems to be  causing it to always be "true" according to the k8s logs, so the test is right.